### PR TITLE
Habtamu Abere ~ CR5 ~ checkpoint-06

### PR DESCRIPTION
**Pull request by HabRonan**
comfort- 4
completeness - 5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,7 +13,10 @@
 Instantiate a new Angular module called `blog` that takes `ui.router` as a dependency.
 
 ```js
-// Your answer goes here...
+angular
+.module ("blog",
+["ui.router"]
+)
```
Comment (amaseda) [inline README.md:19]: :+1:

File: README.md [hunk 2]
```diff
@@ -26,15 +29,26 @@ One button below has an `ng-click` attribute; the other has `data-ng-click` inst
 ```
 
 ```text
-Your answer goes here...
+there is no difference interms of functionality but the only difference is that data-ng-click will pass through an html-validator while ng-click fails
```
Comment (amaseda) [inline README.md:32]: :+1:

File: README.md [hunk 3]
```diff
@@ -26,15 +29,26 @@ One button below has an `ng-click` attribute; the other has `data-ng-click` inst
...
 Which of the three following options demonstrates the best usage of `ng-app`? **Explain your answer.**
 
-```text
-Your answer goes here...
+```
+all three can work without an error.
+It kinda depends on the developer and the scope of the angular application that the developer wants to hava access to.
+but you can't call data-ui-sref outside the scope.
+
+#### A - demonstrates the better  use of ng-app directive for a developer that wants to have access of the directive as the element that ng-app is attached to will define the scope of the control of the angular application.This directive designates the root element of the application and should be placed to the root element of the html page which is near the root element of the page. e.g. <body> or <html>
```
Comment (amaseda) [inline README.md:50]: :+1:

File: README.md [hunk 4]
```diff
@@ -100,7 +114,7 @@ Which one of the following concepts does this best illustrate?
 What is the `ui-sref` directive, and how is it used?
 
 ```text
-Your answer goes here...
+it's basically a link which helps change the view of your single page angular application depending on the state of your angular application without http request. no hard refresh,which makes it seamless for building great user interfaces and faster applications
```
Comment (amaseda) [inline README.md:117]: :+1:

File: README.md [hunk 5]
```diff
@@ -91,7 +105,7 @@ Which one of the following concepts does this best illustrate?
 ```
 [ ] A: Modularity
 [ ] B: MVC
-[ ] C: Two-way data-binding
+[x] C: Two-way data-binding
```
Comment (amaseda) [inline README.md:108]: :+1:

File: README.md [hunk 6]
```diff
@@ -129,7 +154,20 @@ Let's say the Posts in the previous question are available at `http://localhost:
   3. If Step 1 is unsuccessful, print an error message to the console
 
 ```js
-// Your answer goes here...
+$.ajax({
```
Comment (amaseda) [inline README.md:157]: :+1:

File: README.md [hunk 7]
```diff
@@ -142,5 +180,25 @@ Using the same front-end application and Rails API from the previous question, h
...
 ```js
-// Your answer goes here...
+
+
+    $.ajax({
+       type: 'POST',
+       dataType: 'json',
+       url: "http://localhost:3000",
+       data: {
+       post: {
```
Comment (amaseda) [inline README.md:190]: :+1:

File: README.md [hunk 8]
```diff
@@ -118,7 +132,18 @@ end
...
 
 ```rb
-# Your answer goes here...
+class PostsController < ApplicationController
+  def index
+    @posts = Post.all
+
+     respond_to do  |format|
+
+       format.html { render html: @posts}
```
Comment (amaseda) [inline README.md:141]: You can't pass `@posts` into html -- you need to indicate which specific view you want to see (e.g., `:index`).

Comment (HabRonan) [reply README.md:141]: Thanks.

**Pull request closed without merging**